wallet/wallet-rpc-lib/tests/basic.rs

```diff
@@ -104,21 +117,36 @@ async fn send_coins_to_acct1(#[case] seed: Seed) {
     let to_send_amount = Amount::from_atoms(utxo_amount / 2);
     let _: () = {
         let to_send_amount_str =
-            to_send_amount.into_fixedpoint_str(tf.wallet_service.chain_config().coin_decimals());
+            to_send_amount.into_fixedpoint_str(tf.chain_config().coin_decimals());
         let send_to_addr = acct1_addr.address;
         let options = TransactionOptions { in_top_x_mb: 3 };
         let params = (ACCOUNT0_ARG, send_to_addr, to_send_amount_str, options);
         wallet_rpc.request("address_send", params).await.unwrap()
     };
 
     let balances: Balances = wallet_rpc.request("account_balance", [ACCOUNT0_ARG]).await.unwrap();
+
+    tokio::time::sleep(std::time::Duration::from_millis(300)).await;
+    log::info!("{:#}", wallet_events.next().await.unwrap().unwrap());
+    log::info!("{:#}", wallet_events.next().await.unwrap().unwrap());
+
     let coins_after = balances.coins().to_amount(coin_decimals).unwrap();
     assert!(coins_after <= (coins_before / 2).unwrap());
     assert!(coins_after >= (coins_before / 3).unwrap());
 
     let balances: Balances = wallet_rpc.request("account_balance", [ACCOUNT1_ARG]).await.unwrap();
     log::info!("acct1 balances: {balances:?}");
 
+    let _result: JsonValue = wallet_rpc
+        .request("node_generate_block", (ACCOUNT0_ARG, [(); 0]))
+        .await
+        .unwrap();
```

Comment:
I wanted to check the state of the transaction tracked by the wallet changes when it's in a block. This call does generate a block but it does not hit the chainstate so no further wallet events are generated. Any suggestions how to achieve that?